File: ESAMI/_Testi esami ASD [OFFICIAL]/febbraio-2019/C-190219/graph.c
#include "graph.h"
#include <stdlib.h>
#include <stdio.h>

graph * graph_new(int n_nodi) {
	graph* g = (graph*) malloc(sizeof(graph));
	g->n_vertices = n_nodi;
	g->n_edges = 0;
	g->mat = (double**) malloc(sizeof(double*) * n_nodi);
	for(int i = 0; i < n_nodi; ++i){
		g->mat[i] = (double*) malloc(sizeof(double) * n_nodi);
		for(int j = 0; j < n_nodi; ++j){
			g->mat[i][j] = -1;
		}
	}
	return g;
}

void graph_print(graph* g){
	for(int i = 0; i < g->n_vertices; ++i){
		printf("\t%d\t", i);
	}
	printf("\n");
	for(int i = 0; i < g->n_vertices; ++i){
		printf("%d\t", i);
		for( int j = 0; j < g->n_vertices; ++j){
			printf("%f\t", g->mat[i][j]);
		}
		printf("\n");
	}
	printf("The graph has %d nodes and %d edges.\n", g->n_vertices, g->n_edges);
}

void graph_delete(graph* g){
	for(int i = 0; i < g->n_vertices; ++i){
		free(g->mat[i]);
	}
	free(g->mat);
	free(g);
}

File: ESAMI/Svolti_[OLD_AA]/SVOLTI_DA_ME/DOING/19-02-2019/java-aux-190219/Graph.java
public class Graph {
	
	double[][] mat;
	private int nVertices;
	private int nEdges;
	
	public enum Status {UNEXPLORED, EXPLORED, EXPLORING};
	
	public Graph(int nNodi) {
		this.nVertices = nNodi;
		this.nEdges = 0;
		mat = new double[nNodi][nNodi];
		for (int i = 0; i < nNodi; ++i) {
			for (int j = 0; j < nNodi; ++j) {
				mat[i][j] = -1;
			}
		}
	}
	
	public String toString() {
		StringBuffer sb = new StringBuffer();
		for( int i = 0; i < this.nVertices; ++i) {
			sb.append("\t" + i);
		}
		sb.append("\n");
		for( int i = 0; i < this.nVertices; ++i) {
			sb.append(i + "\t");
			for (int j = 0; j < this.nVertices; ++j) {
				sb.append(mat[i][j] + "\t");
			}
			sb.append("\n");
		}
		sb.append("The graph has " + nVertices + " nodes and " + nEdges + " edges.\n");
		return sb.toString();
	}

	public double[][] getMat() {
		return mat;
	}

	public void setMat(double[][] mat) {
		this.mat = mat;
	}

	public int getnVertices() {
		return nVertices;
	}

	public void setnVertices(int nVertices) {
		this.nVertices = nVertices;
	}

	public int getnEdges() {
		return nEdges;
	}

	public void setnEdges(int nEdges) {
		this.nEdges = nEdges;
	}
	
}
                                  